### DISCUSSION TITLE
Code test coverage?

### DISCUSSION THREAD

**Discussion by m1212e** [viz-rs/vidi#62] [Ideas]
Did you consider or event already use some kind of code coverage tool? I can recommend [tarpaulin](https://github.com/actions-rs/tarpaulin).

**Comment by fundon** [comment]
Yes, need it, and need to add more test cases. [cargo-llvm-cov](https://github.com/taiki-e/cargo-llvm-cov) is also good. 
[tarpaulin](https://github.com/actions-rs/tarpaulin) has not been maintained for too long.

**Comment by fundon** [comment]
```console
$ cargo llvm-cov --html -- --nocapture
$ open target/llvm-cov/html/index.html
```

![  ](https://user-images.githubusercontent.com/27926/208238917-3bbfd77d-d446-4703-853a-8f9f0a337723.png)
